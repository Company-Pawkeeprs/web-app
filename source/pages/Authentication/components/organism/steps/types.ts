export type StepProps = {
    [key: string]: any;
    nextStep: () => void;
    prevStep: () => void;
}
