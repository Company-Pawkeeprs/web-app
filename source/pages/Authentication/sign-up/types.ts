export type InitialStateSignUp = {
    password: string,
    email: string,
    username: string,
    document: string,
}