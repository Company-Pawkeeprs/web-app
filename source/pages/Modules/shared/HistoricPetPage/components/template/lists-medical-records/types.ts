export type ItemProps<G> = {
    item: G
    document: string
    id_pet: string
}
