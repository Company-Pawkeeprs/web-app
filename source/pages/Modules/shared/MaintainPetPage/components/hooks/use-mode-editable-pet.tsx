import { create } from 'zustand'
import type { ModeInput } from '~/Components/molecules/field-control/field-control'

type State = {
    mode: ModeInput
    onChangeMode: (mode: ModeInput) => void
    toggleMode: () => void
}

export const useModeEditablePet = create<State>((set) => ({
    mode: 'editable',
    onChangeMode: (mode) => set({ mode }),
    toggleMode: () =>
        set((state) => ({
            mode: state.mode === 'editable' ? 'readonly' : 'editable',
        })),
}))
