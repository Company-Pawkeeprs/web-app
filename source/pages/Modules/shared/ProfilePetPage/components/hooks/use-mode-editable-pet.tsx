import { create } from 'zustand'
import type { ModeInput } from '~/Components/molecules/field-control/field-control'

type State = {
    mode: ModeInput
    onChangeMode: (mode: ModeInput) => void
}

export const useModeEditablePet = create<State>((set) => ({
    mode: 'editable',
    onChangeMode: (mode) => set({ mode }),
}))
