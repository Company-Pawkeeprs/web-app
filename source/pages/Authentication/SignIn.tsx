import Link from 'next/link';
import React, { FormEvent, useState } from 'react';
import { Button, Card, Col, Container, Input, Label, Row } from 'reactstrap';
import AuthSlider from '~/Components/organism/auth-carousel';
import FooterAuth from '~/Components/organism/footer-auth';

import HeaderTitle from '~/Components/atoms/header-title';
import LOADING from '~/constants/loading';
import { useAuth } from '~/contexts/auth-context';
import { onChangePassword, onChangeUsername, onToggleVisiblePassword } from '~/store/auth/loginV2/slice';
import { useAppDispatch } from '~/store/hooks';

const CoverSignIn = () => {

    const dispatch = useAppDispatch()

    const [remember, setRemember] = useState(false);

    const {
        signIn,
        isLoading,
        password,
        username,
        visiblePassword
    } = useAuth()



    const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
        e.preventDefault()
        signIn({
            username,
            password
        })
    }

    const toggleVisiblePassword = () => {
        dispatch(onToggleVisiblePassword())
    }

    const extractTargetValue = (callback: any) => (e: React.ChangeEvent<HTMLInputElement>) => {
        dispatch(callback(e.target.value))
    }

    const toggleRemember = () => {
        setRemember(!remember)
    }


    const disabled = isLoading === LOADING.PENDING

    return (
        <React.Fragment>
            <HeaderTitle title="Sign In" />
            <div className="auth-page-wrapper auth-bg-cover py-5 d-flex justify-content-center align-items-center min-vh-100">
                <div className="bg-overlay" />

                <div className="auth-page-content overflow-hidden pt-lg-5">
                    <Container>
                        <Row>
                            <Col lg={12}>
                                <Card className="overflow-hidden">
                                    <Row className="g-0">
                                        <AuthSlider bg='auth-bg-image-3' />

                                        <Col lg={6}>
                                            <div className="p-lg-5 p-4">
                                                <div>
                                                    <h5 className="text-primary">Bem Vindo!</h5>
                                                    <p className="text-muted">Entre para ter acesso a todas as funcionalidades.</p>
                                                </div>

                                                <div className="mt-4">
                                                    <form onSubmit={handleSubmit} >

                                                        <div className="mb-3">
                                                            <Label htmlFor="username" className="form-label">Nome do Usuário:</Label>
                                                            <Input
                                                                type="text"
                                                                className="form-control"
                                                                id="username"
                                                                placeholder="Digite seu nome de usuário"
                                                                required
                                                                value={username}
                                                                onChange={extractTargetValue(onChangeUsername)}
                                                            />
                                                        </div>

                                                        <div className="mb-3">
                                                            <div className="float-end">
                                                                <Link href="/forget-password" className="text-muted">Esqueceu a Senha?</Link>
                                                            </div>
                                                            <Label className="form-label" htmlFor="password-input">Senha</Label>
                                                            <div className="position-relative auth-pass-inputgroup mb-3">
                                                                <Input
                                                                    type={visiblePassword ? "text" : "password"}
                                                                    className="form-control pe-5 password-input"
                                                                    placeholder="Digite sua senha"
                                                                    id="password-input"
                                                                    required
                                                                    value={password}
                                                                    onChange={extractTargetValue(onChangePassword)}
                                                                />
                                                                <button
                                                                    className="btn btn-link position-absolute end-0 top-0 text-decoration-none text-muted password-addon"
                                                                    type="button"
                                                                    id="password-addon"
                                                                    onClick={toggleVisiblePassword}
                                                                >
                                                                    {
                                                                        visiblePassword ?
                                                                            <i className="ri-eye-off-fill align-middle"></i> :
                                                                            <i className="ri-eye-fill align-middle"></i>
                                                                    }
                                                                </button>
                                                            </div>
                                                        </div>

                                                        <div className="form-check">
                                                            <Input
                                                                className="form-check-input"
                                                                type="checkbox"
                                                                value=""
                                                                checked={remember}
                                                                onChange={toggleRemember}
                                                                id="auth-remember-check"
                                                            />
                                                            <Label className="form-check-label" htmlFor="auth-remember-check">Lembrar-me</Label>
                                                        </div>

                                                        <div className="mt-4">
                                                            <Button color="success" className="w-100" type="submit" disabled={disabled}>Entrar</Button>
                                                        </div>

                                                        <div className="mt-4 text-center">
                                                            <div className="signin-other-title">
                                                                <h5 className="fs-14 mb-4 title">Logar Com:</h5>
                                                            </div>

                                                            <div>
                                                                {/* <Button color="primary" className="btn-icon me-1"><i className="ri-facebook-fill fs-16"></i></Button> */}
                                                                <Button color="danger" className="btn-icon me-1"><i className="ri-google-fill fs-16"></i></Button>
                                                            </div>
                                                        </div>

                                                    </form>
                                                </div>

                                                <div className="mt-5 text-center">
                                                    <p className="mb-0">Você não tem uma conta ? <Link href="/sign-up" className="fw-bold text-primary text-decoration-underline"> Registre-se</Link> </p>
                                                </div>
                                            </div>
                                        </Col>
                                    </Row>
                                </Card>
                            </Col>
                        </Row>
                    </Container>
                </div>

                <FooterAuth />

            </div>
        </React.Fragment>
    );
};

export default CoverSignIn;