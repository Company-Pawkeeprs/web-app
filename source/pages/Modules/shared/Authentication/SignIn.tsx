import { useEffect } from 'react'

import { LOADING } from '~/constants/loading'

import { useRouter } from 'next/navigation'
import { BtnLink } from '~/Components/atoms/btn'
import { useAppDispatch, useAppSelector } from '~/store/hooks'
import { resetLoading } from '~/store/slices/auth/login/actions'
import AuthLayout from '../../_layouts/auth/auth_layout'
import LoadingPage from '../LoadingPage'
import AuthInputs from './components/organism/auth-inputs'

const CoverSignIn = () => {
    const router = useRouter()
    const { isLoading } = useAppSelector((state) => state.Login)
    const dispatch = useAppDispatch()

    const loading = isLoading === LOADING.PENDING || isLoading === LOADING.SUCCESS

    useEffect(() => {
        if (isLoading === LOADING.SUCCESS) {
            router.prefetch('/dashboard')
            setTimeout(() => {
                router.push('/dashboard')
            }, 1000)
        }
    }, [isLoading])

    useEffect(() => {
        return () => {
            dispatch(resetLoading())
        }
    }, [])

    if (isLoading === LOADING.SUCCESS) {
        return (
            <div className="min-h-screen auth-bg-cover flex flex-col ">
                <div className="bg-overlay" />
                <LoadingPage />
            </div>
        )
    }

    return (
        <AuthLayout title="Entrar" image="/bg-sign-in.webp" alt="Imagem" hasImage>
            <div className="flex flex-col justify-center items-center gap-3 lg:mt-5">
                <p className="text-sm font-bold text-secondary-500">
                    Seja Bem-vindo(a)!
                </p>
            </div>
            <div className="mobile:!mt-0 mobile:p-0 web:p-4">
                {loading && (
                    <div className="flex justify-center item-center web:min-h-[236px]">
                        <div
                            className="spinner-border text-primary-500 w-44 h-44 my-4"
                            role="status"
                        >
                            <span className="visually-hidden">Loading...</span>
                        </div>
                    </div>
                )}

                {!loading && (
                    <div className="web:max-h-[236px]">
                        <AuthInputs />
                    </div>
                )}
            </div>

            <div className="w-full flex flex-col justify-center items-center ">
                <p className="-mb-2 text-gray-400 font-normal">
                    Você não tem uma conta ?
                </p>
                <BtnLink message="Criar Conta" href="/sign-up" />
            </div>
        </AuthLayout>
    )
}

export default CoverSignIn
