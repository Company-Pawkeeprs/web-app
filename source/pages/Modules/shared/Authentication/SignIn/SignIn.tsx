import { LOADING } from '~/constants/loading'

import { BtnLink } from '~/Components/atoms/btn'
import Loader from '~/Components/organism/loader'
import useModeProfile from '~/hooks/use-mode'
import { useAppSelector } from '~/store/hooks'
import { ModeProfile } from '~/types/profile'
import AuthLayout from '../../../_layouts/auth/auth_layout'
import AuthInputs from '../components/organism/auth-inputs'

export type CoverSignInProps = {
    bgImage: '/bg-sign-in.webp' | '/bg-three.jpg' | '/bg-sign-up.webp'
}

const CoverSignIn = ({ bgImage }: CoverSignInProps) => {
    const { mode } = useModeProfile()

    const { isLoading } = useAppSelector((state) => state.Login)

    const isPending = isLoading === LOADING.PENDING
    const isSuccessful = isLoading === LOADING.SUCCESS
    const loading = isPending || isSuccessful

    const link = mode === ModeProfile.vet ? '/veterinary/sign-up' : '/tutor/sign-up'

    return (
        <AuthLayout
            title="Entrar"
            image={bgImage}
            alt="Imagem"
            hasImage
            loading={loading}
        >
            {!loading && (
                <div className="flex flex-col items-center justify-center">
                    <p className="text-sm font-bold text-secondary-500">
                        Seja Bem-vindo(a)!
                    </p>
                </div>
            )}
            <div className="mobile:!mt-0 mobile:p-0 web:p-1">
                <Loader
                    condition={loading}
                    message="Aguarde um momento..."
                    type="TailSpin"
                />

                {!loading && (
                    <div className="web:max-h-[236px] p-2">
                        <AuthInputs mode={mode} />
                    </div>
                )}
            </div>

            {!loading && (
                <div className="flex flex-col items-center justify-center w-full mt-4 ">
                    <p className="font-normal text-gray-400 ">
                        Você não tem uma conta ?
                    </p>
                    <BtnLink message="Criar Conta" href={link} />
                </div>
            )}
        </AuthLayout>
    )
}

export default CoverSignIn
