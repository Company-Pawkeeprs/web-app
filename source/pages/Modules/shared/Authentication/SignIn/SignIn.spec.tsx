import { fireEvent, render, screen } from '@testing-library/react'
import { act } from 'react'
import { describe, expect, it, vi, beforeEach } from 'vitest'
import * as useAuth from '~/hooks/use-auth'
import * as useModeProfile from '~/hooks/use-mode'
import ProviderClient from '~/store'
import SignIn from './SignIn'
import { LOADING } from '~/helpers/loading'
import { ModeProfile } from '~/types/profile'
import * as selector from '~/store/hooks'

const Wrapper = () => (
    <ProviderClient>
        <SignIn bgImage="/bg-sign-in.webp" />
    </ProviderClient>
)

describe('SignIn (Unit)', () => {
    it('deve corresponder ao snapshot', () => {
        const { container } = render(<Wrapper />)
        expect(container).toMatchSnapshot()
    })

    it('deve ter o atributo href para page /tutor/sign-up no modo tutor', async () => {
        vi.spyOn(useModeProfile, 'default').mockReturnValueOnce({
            mode: ModeProfile.tutor,
        })

        render(<Wrapper />)

        const createAccoutButton = screen.getByText('Criar Conta')

        await act(async () => {
            fireEvent.click(createAccoutButton)
        })

        expect(createAccoutButton).toHaveAttribute('href', '/tutor/sign-up')
    })

    it('deve ter o atributo href para page /veterinary/sign-up no modo vet', async () => {
        vi.spyOn(useModeProfile, 'default').mockReturnValueOnce({
            mode: ModeProfile.vet,
        })

        render(<Wrapper />)

        const createAccoutButton = screen.getByText('Criar Conta')

        await act(async () => {
            fireEvent.click(createAccoutButton)
        })

        expect(createAccoutButton).toHaveAttribute('href', '/veterinary/sign-up')
    })

    /*it('deve estar carregando durante a execucao do login', async () => {
        vi.spyOn(selector, 'useAppSelector').mockReturnValue({
            Login: { isLoading: LOADING.PENDING },
        })

        render(<Wrapper />)

        const loaderComponent = screen.getByTestId('loading-indicator')

        expect(loaderComponent).toBeInTheDocument()
    })*/
})
