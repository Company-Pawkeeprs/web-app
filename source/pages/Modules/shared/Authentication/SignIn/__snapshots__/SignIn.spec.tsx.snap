// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignIn (Unit) > deve corresponder ao snapshot 1`] = `
<div>
  <div
    class="flex flex-col min-h-screen auth-bg-cover "
  >
    <div
      class="bg-overlay"
    />
    <main
      class="flex flex-1 content-center mobile:content-start items-center justify-center mobile:items-start mobile:justify-start mobile:overflow-hidden"
    >
      <section
        class="grid mobile:flex-1 mobile:min-h-screen mobile:flex-col mobile:!grid-cols-1 mobile:!h-full mobile:!w-full z-10 shadow-2xl rounded-lg mobile:rounded-none mobile:shadow-none mobile:overflow-hidden mobile:bg-white w-[80%] grid-cols-2"
      >
        <picture
          class="flex mobile:h-64  overflow-hidden !w-full h-full relative rounded-l-xl mobile:rounded-l-none"
        >
          <img
            alt="Imagem"
            class="flex flex-1 w-full h-full bg-no-repeat !bg-cover mobile:!bg-contain"
            src="/bg-sign-in.webp"
          />
          <img
            alt="Imagem"
            class="absolute inset-0 hidden object-fill w-full h-full mobile:block"
            src="/bg-sign-in.webp"
          />
        </picture>
        <div
          class="
                                !overflow-hidden relative mobile:rounded-r-none 
                                mobile:rounded-none grid grid-cols-1 mobile:!w-full 
                                mobile:!min-h-full py-4 px-12 
                                mobile:py-2 mobile:px-4 bg-white
                             rounded-r-xl"
        >
          <div
            class="flex h-40 mobile:h-36 w-full !bg-contain !bg-no-repeat !bg-center"
            style="background: url(/logo-default.webp);"
          />
          <div
            class="flex flex-col items-center justify-center"
          >
            <p
              class="text-sm font-bold text-secondary-500"
            >
              Seja Bem-vindo(a)!
            </p>
          </div>
          <div
            class="mobile:!mt-0 mobile:p-0 web:p-1"
          >
            <div
              class="web:max-h-[236px] p-2"
            >
              <form
                action="#"
                class="w-full p-2"
              >
                <div
                  class="w-full p-1"
                >
                  <label
                    class="text-xs font-semibold text-gray-500 gap-1 text-start items-center label justify-start pb-1 mb-0 h-fit"
                    data-testid="label-username"
                    for="username"
                  >
                    Email:
                  </label>
                  <div
                    class="relative"
                  >
                    <input
                      class="w-full px-4 py-2 duration-100 ease-in-out transition-shadow disabled:opacity-70 text-gray-600 disabled:bg-gray-200 text-base h-11 invalid:!shadow-outline-secondary-500 invalid:!text-secondary-500 invalid:border-secondary-500 invalid:focus:border-gray-300 border font-sans dark:bg-[#292e33] dark:text-gray-200 focus:ring-2 focus:ring-primary-500 bg-white focus:!border-none focus:!ring-2 focus:!ring-primary-500 outline-none rounded-md overflow-visible"
                      ctx="[object Object]"
                      data-testid="email-input"
                      id="username"
                      name="username"
                      pattern="[^\\s]+"
                      placeholder="Digite seu email"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="flex flex-col items-end justify-center w-full mb-3 position-relative"
                >
                  <div
                    class="w-full p-1"
                  >
                    <label
                      class="text-xs font-semibold text-gray-500 gap-1 text-start items-center label justify-start pb-1 mb-0 h-fit"
                      data-testid="label-password"
                      for="password"
                    >
                      Senha:
                    </label>
                    <div
                      class="relative"
                    >
                      <div
                        class="absolute inset-y-0 flex items-center pl-1 text-sm text-gray-400 pointer-events-none left-1"
                      >
                        <svg
                          aria-hidden="true"
                          class="w-5 h-5 "
                          data-slot="icon"
                          fill="none"
                          stroke="currentColor"
                          stroke-width="1.5"
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M15.75 5.25a3 3 0 0 1 3 3m3 0a6 6 0 0 1-7.029 5.912c-.563-.097-1.159.026-1.563.43L10.5 17.25H8.25v2.25H6v2.25H2.25v-2.818c0-.597.237-1.17.659-1.591l6.499-6.499c.404-.404.527-1 .43-1.563A6 6 0 1 1 21.75 8.25Z"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                          />
                        </svg>
                      </div>
                      <input
                        class="w-full px-4 py-2 duration-100 ease-in-out transition-shadow disabled:opacity-70 text-gray-600 disabled:bg-gray-200 text-base h-11 invalid:!shadow-outline-secondary-500 invalid:!text-secondary-500 invalid:border-secondary-500 invalid:focus:border-gray-300 border font-sans dark:bg-[#292e33] dark:text-gray-200 focus:ring-2 focus:ring-primary-500 bg-white focus:!border-none focus:!ring-2 focus:!ring-primary-500 outline-none rounded-md overflow-visible !pl-8 !pr-8"
                        ctx="[object Object]"
                        data-testid="password-input"
                        id="password"
                        name="password"
                        placeholder="Digite sua senha"
                        position="right"
                        type="password"
                        value=""
                      />
                      <div
                        class="absolute right-0 mr-2 transform -translate-y-1/2 top-1/2"
                      >
                        <button
                          aria-label="toggle password visibility"
                          class="flex items-center justify-center opacity-50"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="w-5 h-5"
                            data-slot="icon"
                            fill="none"
                            stroke="currentColor"
                            stroke-width="1.5"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M3.98 8.223A10.477 10.477 0 0 0 1.934 12C3.226 16.338 7.244 19.5 12 19.5c.993 0 1.953-.138 2.863-.395M6.228 6.228A10.451 10.451 0 0 1 12 4.5c4.756 0 8.773 3.162 10.065 7.498a10.522 10.522 0 0 1-4.293 5.774M6.228 6.228 3 3m3.228 3.228 3.65 3.65m7.894 7.894L21 21m-3.228-3.228-3.65-3.65m0 0a3 3 0 1 0-4.243-4.243m4.242 4.242L9.88 9.88"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                            />
                          </svg>
                        </button>
                      </div>
                    </div>
                  </div>
                  <a
                    class="rounded-md text-center hover:brightness-90 transition duration-100 ease-in-out disabled:opacity-50 disabled:cursor-not-allowed bg-opacity-80 hover:bg-opacity-100 active:opacity-100 hover:transform hover:scale-[101%] disabled:hover:scale-100 text-opacity-100 text-secondary-500 dark:text-primary-600 hover:no-underline z-10 flex items-center content-end justify-end p-0 m-0 text-xs font-semibold h-fit w-fit"
                    href="/forgot-password"
                  >
                    Esqueceu a senha?
                  </a>
                </div>
                <button
                  class="mobile:p-4 mobile:w-full w-40 flex-grow px-2 h-11 py-1 text-sm font-semibold rounded-md text-center leading-1 hover:brightness-90 transition duration-100 ease-in-out disabled:opacity-50 disabled:cursor-not-allowed bg-opacity-80 hover:bg-opacity-100 active:opacity-100 hover:transform hover:scale-[101%] disabled:hover:scale-100 text-opacity-100 text-gray-50 flex justify-center items-center bg-primary-500 !w-full"
                  color="primary"
                  data-testid="submit-button"
                  disabled=""
                  id=":r0:"
                  type="submit"
                >
                  Entrar
                </button>
              </form>
            </div>
          </div>
          <div
            class="flex flex-col items-center justify-center w-full mt-4 "
          >
            <p
              class="font-normal text-gray-400 "
            >
              Você não tem uma conta ?
            </p>
            <a
              class="text-sm font-semibold rounded-md text-center leading-1 hover:brightness-90 transition duration-100 ease-in-out disabled:opacity-50 disabled:cursor-not-allowed bg-opacity-80 hover:bg-opacity-100 active:opacity-100 hover:transform hover:scale-[101%] disabled:hover:scale-100 text-opacity-100 text-secondary-500 dark:text-primary-600 hover:no-underline"
              href="/tutor/sign-up"
            >
              Criar Conta
            </a>
          </div>
        </div>
      </section>
    </main>
    <footer
      class="w-full py-2 mobile:hidden z-50 bg-transparent"
    >
      <div
        class="flex flex-row items-center justify-between px-6 mobile:flex-col"
      >
        <div>
          <p
            class="text-xs font-semibold text-gray-200"
          >
            2024
             © PawKeepr Sistema Inteligente para Pets Inova Simples LTDA. 49.173.821/0001-19
          </p>
          <p
            class="mt-2 text-xs font-semibold text-gray-200"
          >
            Rua Riachuelo, 1200, São José, 49015-160 Aracaju.
          </p>
        </div>
        <div
          class="flex flex-row h-full gap-4 "
        >
          <a
            class="rounded-md text-center hover:brightness-90 transition duration-100 ease-in-out disabled:opacity-50 disabled:cursor-not-allowed bg-opacity-80 hover:bg-opacity-100 active:opacity-100 hover:transform hover:scale-[101%] disabled:hover:scale-100 text-opacity-100 dark:text-primary-600 hover:no-underline w-20 text-xs font-semibold text-gray-600 hover:text-secondary-500"
            href="/tutor/privacy-policy"
            target="_blank"
          >
            <span
              class="flex justify-center items-center w-6 h-6"
            >
              Política de Privacidade
            </span>
          </a>
          <a
            class="rounded-md text-center hover:brightness-90 transition duration-100 ease-in-out disabled:opacity-50 disabled:cursor-not-allowed bg-opacity-80 hover:bg-opacity-100 active:opacity-100 hover:transform hover:scale-[101%] disabled:hover:scale-100 text-opacity-100 dark:text-primary-600 hover:no-underline w-20 text-xs font-semibold text-gray-600 hover:text-secondary-500"
            href="/tutor/service-terms"
            target="_blank"
          >
            <span
              class="flex justify-center items-center w-6 h-6"
            >
              Termos de Uso
            </span>
          </a>
        </div>
        <div
          class="flex items-center"
        >
          <a
            aria-label="Instagram"
            class="mx-2 text-gray-600 transition-colors duration-300 hover:text-primary-600"
            href="https://www.instagram.com/pawkeepr.app/"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
              />
            </svg>
          </a>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
